app/src/main/java/ch/epfl/sdp/social/ChatActivity.java

```diff
@@ -45,25 +47,26 @@ protected void onCreate(Bundle savedInstanceState) {
         chatRepo = ChatRepository.createRepo(this);
         chatRepo.setContextActivity(this);
 
-        chatRepo.getChat("stupid1@gmail.com", chattingWith);
+        chat = chatRepo.getChat("stupid1@gmail.com", chattingWith);
         sendButton.setOnClickListener(this::onSendClicked);
 
         // Very important to use the singleton to reduce thinking (Done before activity switch inside friendsListActivity)
         // setRemoteToSQLiteAdapter(new FireStoreToSQLiteAdapter().getInstance()); done in friendsListActivity
         loadExistingMessages();
+        messageAdapter.sortMessages();

```

Comment:
loadExistingMessages is **_asynchronous_**. Meaning that while the fetch for messages is ongoing, the messages inside messageAdapter inside will be sorted. Once the fetched messages arrive (after sortMessages is called), they will be placed without sorting hence breaking the sorted-ness of the list inside messageAdapter.